### ISSUE TITLE
[HOLD for checklist][$1000] Web - 'Hmm... it's not here' page is displayed on visiting profile of new user during request money

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22691]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Click on plus and click on Request money
3. Enter any amount and continue
4. Search for new user eg: [new34566+333@gmail.com](mailto:new34566+333@gmail.com)
5. Select the result and click on continue
6. Click on the avatar and observe that app displays 'Hmm its not here' page in place of user details

## Expected Result:
App should also display user details page on click of avatar

## Actual Result:
After clicking avatar, it displays 'Hmm its not here' page in request money flow

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.39-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/60caea74-e7ab-422c-b332-b8a7b19ca502

https://github.com/Expensify/App/assets/93399543/24cb3c70-9769-463a-a8a7-90a6c59f848c

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689083906726459

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010b086092b3e04fff</li>
        <li>Upwork Job ID: 1679016813331095552</li>
        2023-07-26<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Contributor | 25808235</li>
<li>dhanashree-sawant | Reporter | 25808237</li>
</ul></li>
    </ul>
</details>

**Comment by multijump** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Tooltip displays only avatar and 'Hmm its not here' page is displayed on visiting profile of new user during request money

### What is the root cause of that problem?
1. Tooltip's RC 
We generate participants list from `createOption` in `OptionsListUtils.js`
When get avatar here, `createOption` calls `ReportUtils.getIcons`.
https://github.com/Expensify/App/blob/882026d659629d47cda410d120a00ac2a2cc0c61/src/libs/OptionsListUtils.js#L524
`createOption` pass `personalDetail.login` and `personalDetail.accountID` but actually, there is no parameters for these values.
https://github.com/Expensify/App/blob/882026d659629d47cda410d120a00ac2a2cc0c61/src/libs/ReportUtils.js#L807
If participant is the non-exist user in DB, there is not any report so in this case, follow code bloc execute.
```
    if (_.isEmpty(report)) {
        const fallbackIcon = {
            source: defaultIcon || Expensicons.FallbackAvatar,
            type: CONST.ICON_TYPE_AVATAR,
            name: '',
            id: -1,
        };
        return [fallbackIcon];
    }
```
That's why tooltip doesn't have description.

2. Profile Issue's RC
if participant is the non-exist user in DB, Onyx send the optimisticData with random generated accountId.
When click the participant, it navigate to user profile.
```
    const navigateToUserDetail = (option) => {
        if (!option.accountID) {
            return;
        }
        Navigation.navigate(ROUTES.getProfileRoute(option.accountID));
    };
```
But this `option.accountID` is randomly generated so can't get correct profile data from this fake accountID.

### What changes do you think we should make in order to solve the problem?

1. Tooltip Issue
We can add simply user login and accountID in getIcons function.
```
function getIcons(report, personalDetails, defaultIcon = null, isPayer = false, login = '', accountID = undefined) {
    if (_.isEmpty(report)) {
        const fallbackIcon = {
            source: defaultIcon || Expensicons.FallbackAvatar,
            type: CONST.ICON_TYPE_AVATAR,
            name: login,
            id: accountID,
        };
        return [fallbackIcon];
    }
```
2. Profile Issue
- We need to determine the expected result in this case.
- There is no entry in DB in this case, so can't show profile info properly.

### What alternative solutions did you explore? (Optional)
N/A
**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by greg-schroeder** [comment]
The tooltip part of this is a duplicate of https://github.com/Expensify/App/issues/22485 / https://github.com/Expensify/App/issues/22062

**Comment by greg-schroeder** [comment]
I updated the OP/issue title to focus on the missing details in the user profile

**Comment by greg-schroeder** [comment]
@deetergp seemed to think it was per the discussion here, not sure though: https://expensify.slack.com/archives/C049HHMV9SM/p1689083906726459

**Comment by sobitneupane** [comment]
> Profile page not found
We should use the same way that BE used to generate accountID.
Or We could navigate to detailRoute which will display the detail of the temporary option even it exists or not instead of profileRoute

@dukenv0307 What is preventing the ProfilePage to show users with accountID -1 ? Can we update ProfilePage to show those users?

**Comment by sobitneupane** [comment]
Can we pass some fallback data for such cases where BE is unable to send data?

**Comment by sobitneupane** [comment]
Started a discussion: https://expensify.slack.com/archives/C01GTK53T8Q/p1689247696251109

**Comment by multijump** [comment]
> > Profile page not found
> > We should use the same way that BE used to generate accountID.
> > Or We could navigate to detailRoute which will display the detail of the temporary option even it exists or not instead of profileRoute
> 
> @dukenv0307 What is preventing the ProfilePage to show users with accountID -1 ? Can we update ProfilePage to show those users?

I think it is not good to update the ProfilePage to show users with accountID -1.
It would be good to update accountID -1 to optimistic accountID.

**Comment by greg-schroeder** [comment]
Discussion appears to be ongoing

**Comment by greg-schroeder** [comment]
Discussion is still going on - last update was yesterday

**Comment by greg-schroeder** [comment]
Did we get to a point of agreement @sobitneupane?

**Comment by greg-schroeder** [comment]
Thoughts on the updated solution from @dukenv0307 @sobitneupane?

**Comment by sobitneupane** [comment]
Discussion is heading towards not showing Profile at all for users with optimistic AccountID. It will involve change in two cases:

- Creating new report
- IOU operation

**Comment by sobitneupane** [comment]
I don't think so. Can you raise if that's the case in discussion?

**Comment by lschurr** [comment]
Hi @sobitneupane - is this new GH a dupe? https://github.com/Expensify/App/issues/23662

**Comment by sobitneupane** [comment]
@lschurr Yup. Root cause seems similar. Let's hold the other issue. We will probably not show the Profile Page for optimistic users.

**Comment by sobitneupane** [comment]
Thanks for the proposal @dukenv0307.

Your [proposal](https://github.com/Expensify/App/issues/22691#issuecomment-1631954206) looks good to me.

As per the [discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1689247696251109) in slack, we will probably not show the Profile Page at all for the Optimistic AccountID. So, let's make sure we do the same from all places.

Also, please make sure to verify [this issue](https://github.com/Expensify/App/issues/23662) also gets resolved. 

🎀 👀 🎀 C+ reviewed

**Comment by Julesssss** [comment]
Agree with the plan, but we should also make sure the cursor accurately represents whether the avatar can be clicked or not.

**Comment by Julesssss** [comment]
@dukenv0307

**Comment by greg-schroeder** [comment]
Work on linked PR continues

**Comment by greg-schroeder** [comment]
PR still going through testing

**Comment by greg-schroeder** [comment]
Issue Participants:

Issue reported by: @dhanashree-sawant
Contributor: @dukenv0307 
C+: @sobitneupane 

There was a regression here: https://github.com/Expensify/App/issues/22691#issuecomment-1664415375

**Comment by greg-schroeder** [comment]
Sent payments/offers. Remaining payments will be done via NewDot. All that's left is the checklist @sobitneupane

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/20746

> * [ ]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/20746#issuecomment-1685753092

> * [ ]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not requried.

> * [ ]  [@sobitneupane] Determine if we should create a regression test for this bug.
Yes

> * [ ]  [@sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

1. Login with any account
2. Go offline
3. Create a new chat with the user that isn't a recent contact
4. In the report, click the header
5. Verify that the cursor is disabled and can not open profile page
6. Click on the display of the user in welcome message text
7. Verify that the cursor is text and can not open profile page
8. Create a group chat with the user that isn't a recent contact
9. In the report, click the header to open participant page
10. Click on the avatar of this user
11. Verify that the cursor is disable and can not open profile page
12. Go online and verify that the cursor is changed to pointer and can open profile page
13. Go to FAB > request money
14. Enter any amount to go to the next step
15. Search and select the user that isn't a recent contact
16. In confirm page, verify that the cursor is disabled when hovering over the avatar and can not open profile page

Do we agree 👍 or 👎

**Comment by sobitneupane** [comment]
Requested payment on newDot

https://github.com/Expensify/App/issues/22691#issuecomment-1681380731

**Comment by greg-schroeder** [comment]
Thanks @sobitneupane! Took care of payments, filing regression test, closing.

**Comment by JmillsExpensify** [comment]
Reviewed the details for @sobitneupane. $500 approved for payment in NewDot based on BZ summary.
